core/src/consensus/stake/actions.rs

```diff
@@ -102,12 +109,42 @@ impl Action {
                     })?;
                 }
             }
+            Action::ReportDoubleVote {
+                message1,
+                message2,
+            } => {
+                // what is the general verifying for general message?
+                let message_for_signature = message1.message_for_signature();
+                let message_hash = blake256(&message_for_signature);
+
+                let signature1 = message1.signature();
+                let signature2 = message2.signature();
+
+                let signer_idx1 = message1.signer_index();
+                let signer_idx2 = message2.signer_index();
+
+                let signers = signers.ok_or_else(|| {
+                    SyntaxError::InvalidCustomAction(String::from(
+                        "ReportDoubleVote Custom action is invalid for consensus engines without validators",
+                    ))
+                })?;
+                let signer1 = signers
+                    .get(signer_idx1)
```

Comment:
The signer may be different from the message's signer. We should get the prev term's signers.

Reply:
Fixed